crates/runtime/src/instance.rs

```diff
@@ -378,11 +412,21 @@ impl Instance {
     /// Returns `None` if memory can't be grown by the specified amount
     /// of pages.
     pub(crate) fn memory_grow(&self, memory_index: DefinedMemoryIndex, delta: u32) -> Option<u32> {
-        let result = self
+        let memory = self
             .memories
             .get(memory_index)
-            .unwrap_or_else(|| panic!("no memory for index {}", memory_index.index()))
-            .grow(delta);
+            .unwrap_or_else(|| panic!("no memory for index {}", memory_index.index()));
+
+        if let Some(limiter) = &self.limiter {
+            let current = memory.size();
+            let desired = current.checked_add(delta)?;
+
+            if !limiter.memory_growing(current, desired, memory.maximum()) {
+                return None;
+            }
+        }
```

Comment:
Would it be possible to store the `limiter` into the `Memory` itself rather than in the `Instance`?

Reply:
Can do.